packages/apollo/src/utils/lists.ts

```diff
@@ -86,7 +86,18 @@ export class ListHelper {
             return undefined
         }
 
-        return existing.slice(skip, skip + first)
+        const currentList = existing.slice(skip, skip + first).filter(Boolean)
+
+        const cache = options?.cache.extract()
+
+        const hasBrokenLinks = currentList.some((link) => {
+            const ref = (link as Reference)?.__ref
```

Comment:
It's better to avoid type casting here if it has `unknown` type...

Reply:
I would prefer something like

if (typeof item === 'object' && item !== null && item.hasOwnProperty('__ref') and typeof item.__ref === 'string')